### PR TITLE
fix: allow safe apps to disable off-chain signing

### LINKED ISSUE #2440: Enable dapps to disable off-chain signing through `safe_setSettings`
## Bug description
We added a custom RPC call `safe_setSettings` to overwrite the signing behaviour (off-chain or on-chain).
This setting is currently not considered for signing requests from safe apps.

## Environment
any

## Steps to reproduce
 1. Use the test dapp
 2. disable off-chain signing
 3. sign a message

## Expected result
Off-chain signing flow is being used

## Obtained result
On-chain signing flow is being used

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves

Resolves #2440 

## How this PR fixes it
Considers the safe signing settings when signing a message from safe apps.

## How to test it
1. Use the eip1271 test dapp
2. disable off-chain signing
3. sign a message
4. Observe on-chain signature modal

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Comment by katspaugh** [comment]
@schmanu can you give an example for how a Safe App can toggle this? Or is it the user who should toggle it?

**Comment by schmanu** [comment]
> @schmanu can you give an example for how a Safe App can toggle this? Or is it the user who should toggle it?

@katspaugh It can be toggled by a custom RPC call `safe_setSettings`.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -67,9 +67,9 @@ jobs:
       # Extract branch name
       - name: Extract branch name
         shell: bash
-        ## Cut off "refs/heads/" and only allow alphanumeric characters,
+        ## Cut off "refs/heads/", only allow alphanumeric characters and convert to lower case,
         ## e.g. "refs/heads/features/hello-1.2.0" -> "features_hello_1_2_0"
-        run: echo "branch=$(echo $GITHUB_HEAD_REF | sed 's/refs\/heads\///' | sed 's/[^a-z0-9]/_/ig')" >> $GITHUB_OUTPUT
+        run: echo "branch=$(echo $GITHUB_HEAD_REF | sed 's/refs\/heads\///' | sed 's/[^a-z0-9]/_/ig') | sed 's/[A-Z]/\L&/g'" >> $GITHUB_OUTPUT
```
Comment (iamacook) [inline .github/workflows/deploy.yml:72]: Is this a requirement for this PR?

Comment (schmanu) [reply .github/workflows/deploy.yml]: Yes I needed it as this PR also contained camel case.

**Review by iamacook — APPROVED** [review]

**Comment by francovenica** [comment]
So the onchain/offchain works, in the sense that if you choose how the message must be signed in the eip app that's what you get in the web app

I still got a bunch  of other issues that I'm not sure are related to the ticket, so I'd like to debug them in a call if possible.

The messages are not showing in the message tab after signing
Trying to sign with a 2nd owner in a safe 2/x is throwing errors
image

**Comment by schmanu** [comment]
@francovenica 
I can reproduce those issues on staging CGW.
When I use the prod CGW this does not happen.
In staging I also do not see the proposed message in the messages tab at all.

**Comment by francovenica** [comment]
Yes. I tried the same and it seems something is wrong with the stg CGW. I posted a [message](https://5afe.slack.com/archives/C049H57FS1E/p1693872003796939) for the infra team to see if something has changed in the past week.

Taking that aside the ticket is fine
The message is being proposed as offchain/onchain depending on the "Enable/Disable off chain signing" in the app itself
image

Made sure that the checkbox in Settings > Safe apps, that forces the message to be on chain no matter what still works as intended

Looks good to me.
I'll re check this ticket after having a response from Infra team

**Comment by katspaugh** [comment]
@francovenica can we merge this, or do you want to check it again?

**Comment by francovenica** [comment]
Give it a 2nd try now that the issue with the CGW was solved

Looks good, I can enable now the on-chian type of message signing in the app and it works fine. Off chain message signing still works fine and the settings in the app that forces on-chain type of singature also works fine as well.

**Pull request merged**